packages/core/src/theme/components.ts

```diff
@@ -9,25 +15,45 @@ type GetConfigProperty<C extends Component> = {
   merge?: boolean;
   variant?: Variant<C>;
   property: keyof ComponentConfig;
+  state?: State<C>;
 };
 
-function get<C extends Component>(
-  name: C,
-  variant?: Variant<C>,
-  merge?: boolean,
-) {
+function get<C extends Component>({
+  name,
+  variant,
+  merge,
+  state,
+}: {
+  name: C;
+  variant?: Variant<C>;
+  merge?: boolean;
+  state?: State<C>;
+}) {
   const config = theme.getValue('components', name);
 
   if (!variant || !config) {
+    if (state) {
+      return {
+        ...config,
+        style: deepMerge(config.style, config.states[state]),
+      };
+    }
```

Comment:
Why is this here? What I mean is:
This block is supposed to be a guard:
if the variant is not defined or if there is no configuration in the theme for that component, then return `config`.
It was correct because, before your changes, if there was no variant then there was no need to merge the base component configuration (`config`) with the variant configuration, and in case the component configuration was not presented in the first place, then for sure, there is also no variant so we can safely return undefined.

Now it seems like we are saying: if there is no component configuration it's possible that there is a state for that component, that does not make sense in my opinion.

Other than that, In case `config` is `undefined` there will be an error:

```typescript
    if (state) {
      return {
        ...config,
        // trying to access style of undefined
        style: deepMerge(config.style, config.states[state]),
      };
    }
```

My suggestion is to change the whole block like this:

```typescript
if (!config) {
    return config;
  }

  if (!variant) {
    if (state) {
      return {
        ...config,
        style: deepMerge(config.style, config.states[state]),
      };
    }
    return config;
  }
```

What do you think about it?